src/backend/distributed/worker/worker_merge_protocol.c

```diff
@@ -53,6 +57,50 @@ static void CopyTaskFilesFromDirectory(StringInfo schemaName, StringInfo relatio
 PG_FUNCTION_INFO_V1(worker_merge_files_into_table);
 PG_FUNCTION_INFO_V1(worker_merge_files_and_run_query);
 PG_FUNCTION_INFO_V1(worker_cleanup_job_schema_cache);
+PG_FUNCTION_INFO_V1(worker_create_schema);
+PG_FUNCTION_INFO_V1(worker_remove_jobdir);
+
+
+/*
+ * worker_create_schema creates a schema with the given job id in local.
+ */
+Datum
+worker_create_schema(PG_FUNCTION_ARGS)
+{
+	uint64 jobId = PG_GETARG_INT64(0);
+
+	StringInfo jobSchemaName = JobSchemaName(jobId);
+	CheckCitusVersion(ERROR);
+
+	bool schemaExists = JobSchemaExists(jobSchemaName);
+	if (!schemaExists)
+	{
+		CreateJobSchema(jobSchemaName);
+	}
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * worker_remove_jobdir removes the job directory with the given job id.
+ */
+Datum
+worker_remove_jobdir(PG_FUNCTION_ARGS)
+{
+	uint64 jobId = PG_GETARG_INT64(0);
+	StringInfo jobDirectoryName = JobDirectoryName(jobId);
+	StringInfo jobSchemaName = JobSchemaName(jobId);
+
+	CheckCitusVersion(ERROR);
+
+	Oid schemaId = get_namespace_oid(jobSchemaName->data, false);
+
+	EnsureSchemaOwner(schemaId);
+	CitusRemoveDirectory(jobDirectoryName);
```

Comment:
We should probably do RemoveJobSchema here as well

Reply:
We use the schemas in the SQL_TASK queries so we should remove them only at the end I think.

For example for `select count(*) from ab k, ab l where k.a = l.b;`
```
LOG:  issuing SELECT count(*) AS count FROM (pg_merge_job_469577498631.task_000033 "pg
_merge_job_469577498631.task_000033" JOIN pg_merge_job_469577498632.task_000033 "pg_me
rge_job_469577498632.task_000033" ON (("pg_merge_job_469577498631.task_000033".interme
diate_column_469577498631_0 OPERATOR(pg_catalog.=) "pg_merge_job_469577498632.task_000
033".intermediate_column_469577498632_0))) WHERE true
```

Reply:
I guess we should do cleanup after the whole execution not after `executeDependentTasks`. Then this wouldnt be a problem